Tasks/KubernetesV1/src/kubernetes.ts

```diff
@@ -82,14 +100,74 @@ function executeKubectlCommand(clusterConnection: ClusterConnection, command: st
         "TaskEndpointId",
         "KubernetesV1",
         JSON.stringify(telemetry));
-    var result = "";
-    return commandImplementation.run(clusterConnection, command, (data) => result += data)
-    .fin(function cleanup() {
-        var commandOutputLength = result.length;
-        if (commandOutputLength > environmentVariableMaximumSize) {
-            tl.warning(tl.loc("OutputVariableDataSizeExceeded", commandOutputLength, environmentVariableMaximumSize));
-        } else {
-            tl.setVariable('KubectlOutput', result);
-        }
-    });
+
+    // The output result can contain more than one Json objects
+    // We want to parse each of the objects separately, hence push the output in JSON array form    
+    var result = [];
+    return commandImplementation.run(clusterConnection, command, (data) => result.push(data))
+        .fin(function cleanup() {
+            console.log("commandOutput" + result);
+            var commandOutputLength = result.length;
+            if (commandOutputLength > environmentVariableMaximumSize) {
+                tl.warning(tl.loc("OutputVariableDataSizeExceeded", commandOutputLength, environmentVariableMaximumSize));
+            } else {
+                tl.setVariable('KubectlOutput', result.toString());
+            }
+
+            // The deployment data is pushed to evidence store only for commands like 'apply' or 'create' which support Json and Yaml output format
+            if (publishPipelineMetadata && publishPipelineMetadata.toLowerCase() == "true" && isJsonOrYamlOutputFormatSupported(command)) {
+                // For each output, check if it contains a JSON object
+                result.forEach(res => {
+                    let parsedObject: any;
+                    if (IsJsonString(res)) {
```

Comment:
if 'outputFormat' argument not specified yaml parse throws exception. We need to check 'outputFormat' specified or not.

Reply:
outPut format is set by default to JSON . User can change the selection via picklist and choose YAML but in any case, the output format will always be present